core/src/style/polylineStyle.cpp

```diff
@@ -339,6 +340,8 @@ void PolylineStyleBuilder<V>::addFeature(const Feature& _feat, const DrawRule& _
             addMesh(line, params);
         }
     } else {
+        params.closedPolygon = true;
```

Comment:
Resolved IRL: A "closed polygon" flag is used in the JS line building code and is set according to the calling function, not the data itself. So if a "line" feature happened to have the same first and last points, JS would still build two caps (as opposed to inferring the line to be closed). Therefore we *should* use a flag like this to match the behavior in JS ... for now.